packages/blaze-tx/src/tx.ts

```diff
@@ -955,13 +955,21 @@ export class TxBuilder {
       // Initialize a CBOR writer to encode the script data.
       const writer = new CborWriter();
       // Encode redeemers and datums into CBOR format.
-      if (redeemers.length === 0) {
-        // An empty redeemer set is always an empty map, as of conway
-        writer.writeStartMap(0);
+      const conway = this.params.protocolVersion.major === 9;
+      if (conway) {
```

Comment:
you can simplify this to just `if (conway && redeemers.length === 0)`, since the two `else` clauses are identical